core/gaspool.go

```diff
@@ -14,12 +14,33 @@
 // You should have received a copy of the GNU Lesser General Public License
 // along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
 
-package types
+package core
 
-import "github.com/ethereum/go-ethereum/core/vm"
+import "math/big"
 
-type BlockProcessor interface {
-	Process(*Block) (vm.Logs, Receipts, error)
-	ValidateHeader(*Header, bool, bool) error
-	ValidateHeaderWithParent(*Header, *Header, bool, bool) error
+// GasPool tracks the amount of gas available during
+// execution of the transactions in a block.
+// The zero value is a pool with zero gas available.
+type GasPool big.Int
+
+// AddGas makes gas available for execution.
```

Comment:
Not really related to this PR but just something that came to mind since it showed up in this diff: with `AddGas` and `SubGas` actually returning a new big.Int and leaves the old one unmodified, should they really be methods?

Since the returned value must always be used for the addition to take effect, making it a function could make it easier to read:
`gp := new(GasPool).AddGas(block.GasLimit())`
vs
`gp := AddGas(gp, block.GasLimit())`